controllers/cluster_restore.go

```diff
@@ -148,6 +168,38 @@ func getOrphanPVCs(
 	return orphanPVCs, nil
 }
 
+func ensureOrphanPodsAreDeleted(ctx context.Context, c client.Client, cluster *apiv1.Cluster) error {
+	var podList corev1.PodList
+	if err := c.List(
+		ctx,
+		&podList,
+		client.InNamespace(cluster.Namespace),
+		client.MatchingLabels{utils.ClusterLabelName: cluster.Name},
```

Comment:
should match label `cnpg.io/podRole` to `instance` to filter out  pooler pod

Reply:
Hey Yanni, thanks, this poses the question: how would you handle the orphan pooler pod if it exists?